rclcpp_lifecycle/test/test_lifecycle_service_client.cpp

```diff
@@ -355,3 +356,53 @@ TEST_F(TestLifecycleServiceClient, lifecycle_transitions) {
   transitions = lifecycle_client()->get_available_transitions();
   EXPECT_EQ(transitions.size(), 0u);
 }
+
+
+TEST_F(TestLifecycleServiceClient, get_service_names_and_types_by_node)
+{
+  auto node1 = std::make_shared<LifecycleServiceClient>("client1");
+  auto node2 = std::make_shared<LifecycleServiceClient>("client2");
+
+  const auto * node_graph =
```

Comment:
Also no need for dynamic_cast here.
```suggestion
  const auto node_graph = node1->get_node_graph_interface();
```